From 0e8941d4f886d40ae9acd8af27f54c4e61478899 Mon Sep 17 00:00:00 2001
From: Zeng Zhaoming <b32542@freescale.com>
Date: Mon, 18 Jul 2011 11:55:41 +0800
Subject: ENGR00152287-3 SDMA: Fix sdma cache bufferable issue

MX6Q desclare dma memory bufferable, which cause sdma
load context failed in 60% possibility.

To fix it, we need to add dsb to flush write buffer before
start dma transfer.

Signed-off-by: Zeng Zhaoming <b32542@freescale.com>
---
 drivers/dma/imx-sdma.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 0ed6c6e..3642ca4 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -373,6 +373,7 @@ static int sdma_run_channel(struct sdma_channel *sdmac)
 
 	init_completion(&sdmac->done);
 
+	wmb();
 	__raw_writel(1 << channel, sdma->regs + SDMA_H_START);
 
 	ret = wait_for_completion_timeout(&sdmac->done, HZ);
@@ -768,6 +769,7 @@ out:
 
 static void sdma_enable_channel(struct sdma_engine *sdma, int channel)
 {
+	wmb();
 	__raw_writel(1 << channel, sdma->regs + SDMA_H_START);
 }
 
-- 
1.7.9.5

